cuebot/src/main/java/com/imageworks/spcue/util/FrameSet.java

```diff
@@ -44,6 +48,140 @@ public int index(int idx) {
+     */
+    private String pad(int number, int width){
+        return Strings.padStart(String.valueOf(number), width, '0');
+    }
+
+    /**
+     * Return a String representation of a frame range based on a list of literal integer frame IDs.
+     * @param frames  A list of integers representing frame IDs,
+     * @return        A representation of a frameset, e.g. '1-10,12-100x2'
+     */
+    private String framesToFrameRanges(ImmutableList frames) {
+        return framesToFrameRanges(frames, 0);
+    }
+
+    /**
+     * Return a String representation of a frame range based on a list of literal integer frame IDs.
+     * @param frames  A list of integers representing frame IDs,
+     * @param zFill   The required width of each left-padded frame ID, e.g. 4 if ID '1' should become '0001'
+     * @return        A representation of a frameset, e.g. '1-10,12-100x2'
+     */
+    private String framesToFrameRanges(ImmutableList<Integer> frames, int zFill) {
+
+        int l = frames.size();
+        if (l == 0) {
+            return "";
+        } else if (l == 1) {
+            return pad(frames.get(0), zFill);
+        }
+
+        ArrayList<String> resultBuilder = new ArrayList<String>();
+
+        int curr_count = 1;
+        int curr_stride = 0;
+        int new_stride = 0;
+        int curr_start = frames.get(0);
+        int curr_frame = frames.get(0);
+        int last_frame = frames.get(0);
+
+        for (int i = 1; i < frames.size(); i++) {
+            curr_frame = frames.get(i);
+
+            if (curr_stride == 0) {
+                curr_stride = curr_frame - curr_start;
+            }
+            new_stride = curr_frame - last_frame;
+            if (curr_stride == new_stride) {
+                last_frame = curr_frame;
+                curr_count += 1;
+                continue;
+            } else if (curr_count == 2 && curr_stride != 1) {
+                resultBuilder.add(pad(curr_start, zFill));
+                curr_stride = 0;
+                curr_start = last_frame;
+                last_frame = curr_frame;
+                continue;
+            } else {
+                resultBuilder.add(buildFrangePart(curr_start, last_frame, curr_stride, zFill));
+                curr_stride = 0;
+                curr_start = curr_frame;
+                last_frame = curr_frame;
+                curr_count = 1;
+                continue;
+            }
+        }
+        if (curr_count == 2 && curr_stride != 1) {
+            resultBuilder.add(pad(curr_start, zFill));
+            resultBuilder.add(pad(curr_frame, zFill));
+        } else {
+            resultBuilder.add(buildFrangePart(curr_start, curr_frame, curr_stride, zFill));
+        }
+
+        StringBuilder sb = new StringBuilder();
+        l = resultBuilder.size()-1;
+        for (int j = 0; j < l; j++) {
+            sb.append(resultBuilder.get(j)).append(",");
+        }
+        sb.append(resultBuilder.get(l));
+        return sb.toString();
```

Comment:
Here's a slight rewrite of this block which uses Java 8 streams and a `StringJoiner` to handle adding the delimiters.

```
StringJoiner sj = new StringJoiner(",");
resultBuilder.forEach(sj::add);
return sj.toString();
```

Reply:
Ah thanks I was was looking for something like this.